### DISCUSSION TITLE
Can we send byte array to Gotenberg

### DISCUSSION THREAD

**Discussion by muhyilmaz** [gotenberg/gotenberg#481] [Q&A] [answered]
I am trying to convert my bytearray docx file to pdf. Is there any option to send my docx file as byte array to your API. I cannot give path to API, I have just byte array data of my file/files. 

Note: I am using dotnet core 6, c# ...

**Comment by gulien** [comment]
Hello @muhyilmaz,

There is no need to send a file path. Just use your favorite C# HTTP client, create a `multipart/form-data` request with a form file (it should be able take a byte array as value), and you're good to go!

You may also want to check the code from https://github.com/Jaben/GotenbergSharpApiClient.

**Reply by muhyilmaz** [reply]
```
public async Task<string> GetPdfFromGotenberg(byte[] file, string ext)
        {
          
            using (HttpClient gothClient = new HttpClient())
            {
                gothClient.BaseAddress = new System.Uri("http://localhost:3000");
                var apiUri = "/forms/libreoffice/convert";
                 
                var byteArrayContent = new ByteArrayContent(file);

                //create a System.Net.Http.MultiPartFormDataContent
                var multipartContent = new MultipartFormDataContent();
                multipartContent.Add(byteArrayContent, $"file.{ext}");
                var result = await gothClient.PostAsync(apiUri, multipartContent);
                var resMessage = result.Content.ReadAsStringAsync();
            }
            return "file";//useless return
        }
```

        
        
I tried this method but it failed. Here is my result: 
![resim_2022-08-10_160623776](https://user-images.githubusercontent.com/34422587/183908951-deaf96e8-6d2e-4fb8-a809-dcd37dd7b6dc.png)

**Reply by onedram** [reply]
Just use another method of multipartContent:
  not 'Add(HttpContent content, string name)'
but this one: 
  Add(HttpContent content, string name, string fileName);

Change
  multipartContent.Add(byteArrayContent, $"file.{ext}")
to
  multipartContent.Add(byteArrayContent, "file", $"file.{ext}")

**Reply by muhyilmaz** [reply]
> Just use another method of multipartContent: not 'Add(HttpContent content, string name)' but this one: Add(HttpContent content, string name, string fileName);
> 
> Change multipartContent.Add(byteArrayContent, $"file.{ext}") to multipartContent.Add(byteArrayContent, "file", $"file.{ext}")

This worked for me. Thanks

**Comment by muhyilmaz** [comment, marked as answer]
Just use another method of multipartContent:
not 'Add(HttpContent content, string name)'
but this one:
Add(HttpContent content, string name, string fileName);

Change
multipartContent.Add(byteArrayContent, $"file.{ext}")
to
multipartContent.Add(byteArrayContent, "file", $"file.{ext}")

@onedram @gulien   thanks for answer
